### PR TITLE
Bump stable structures

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We would like to use the variable memory sizes introduced in `ic-stable-structures 0.6.0`

# Changes
- Update `stable-structures` to version `0.6.0`
- Update the code to match:
  - Instead of having a separate `BoundedStorable` trait, `stable-structures` now has a `BOUND` in the `Storable` trait.

# Tests
Existing tests should suffice.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

**Review by lmuntaner — APPROVED** [review]
Rubber stamp! Thanks!

File: rs/backend/src/accounts_store/schema/accounts_in_stable_memory.rs [hunk 1]
```diff
@@ -157,6 +157,10 @@ pub struct AccountStorageKey {
     bytes: [u8; AccountStorageKey::SIZE],
 }
 impl Storable for AccountStorageKey {
+    const BOUND: Bound = Bound::Bounded {
```
Comment (peterpeterparker) [inline rs/backend/src/accounts_store/schema/accounts_in_stable_memory.rs:160]: Out of curiosity, why not using `Unbounded`?

Comment (bitdivine) [reply rs/backend/src/accounts_store/schema/accounts_in_stable_memory.rs:160]: Unbounded is really new.  Planning to get all the other code ready, buying time to test unbounded and let other projects test it as well, then we can decide at the last moment which we trust more.

Comment (peterpeterparker) [reply rs/backend/src/accounts_store/schema/accounts_in_stable_memory.rs:160]: Isn't Unbounded already used on mainnet by some ckBTC canister?

**Pull request merged**